pkg/client/cache/reflector.go

```diff
@@ -161,7 +161,8 @@ func (r *Reflector) listAndWatch(stopCh <-chan struct{}) {
 			default:
 				glog.Errorf("Failed to watch %v: %v", r.expectedType, err)
 			}
-			return
+			time.Sleep(time.Second)
```

Comment:
@bprashanth This helps in a situation where apiserver restarts (for any reason). Previously all watchers (e.g. 100 kubelets) will immediately send huge number of requests (in 100 node cluster this will be 300 requests just from nodes - 1 from each kube-proxy and 2 from each kubelet).

@lavalamp If we return from this function we will just relist all objects and reenter this one. What kind of errors would be fixed by this, except for "out of window"?